### PR TITLE
✨(forum) Allow sorting forums on the home page

### PR DESCRIPTION
**Pull request by carofun**
## Purpose

Courses can have multiple forums and the current order is not
relevant. The view listing the different forums is by default
ordered by creation, the first forum created is shown first.
It’s interesting to be able to sort forums from other parameters
like alphabetical order, number of topics, number of messages,
or date of last message answered.

Resolve https://github.com/openfun/ashley/issues/233

## Proposal

Order by columns

https://user-images.githubusercontent.com/76939263/148242759-435be2ab-c208-4d4c-acf6-506891fc6358.mov

### CODE REVIEW

**Review by sampaccoud — COMMENTED** [review]
Awesome feature and well executed :star_struck: 

I have proposed some simplification on the Mixin stuff :wink: 
I'm available if you want to discuss some patterns.

**Comment by carofun** [comment]
> Awesome feature and well executed 🤩
> 
> I have proposed some simplification on the Mixin stuff 😉 I'm available if you want to discuss some patterns.

Cool many thank for your precious technical return

File: src/ashley/machina_extensions/forum/views.py [hunk 1]
```diff
@@ -5,34 +5,52 @@
...
+        self.params = {}
+
+    def get(self, request, *args, **kwargs):
+        """Handles GET requests."""
+        # collect the current params for the order by in request
+        self.params = dict(request.GET.items())
+        try:
+            super_get = getattr(super(), "get")
+            return super_get(request, *args, **kwargs)
+        except Exception as exception:
```
Comment (sampaccoud) [inline src/ashley/machina_extensions/forum/views.py:48]: I think you should catch only the precise exception you're expecting

**Review by sampaccoud — APPROVED** [review]

File: src/ashley/machina_extensions/forum/views.py (hunk 1 above)
Comment (carofun) [reply src/ashley/machina_extensions/forum/views.py]: j'ai mis except AttributeError as exception:

**Pull request merged**